server/opts.go

```diff
@@ -1518,6 +1520,10 @@ func parseJetStream(v interface{}, opts *Options, errors *[]error, warnings *[]e
 				opts.JetStreamMaxMemory = mv.(int64)
 			case "max_file_store", "max_file":
 				opts.JetStreamMaxStore = mv.(int64)
+			case "domain":
+				opts.JetStreamDomain = mv.(string)
```

Comment:
We are using this in a subject, can we check that opts.JetStreamDomain contains neither `.` nor ` ` (space) and error otherwise?

Reply:
Dots are actually ok, does not matter. But could check if its a valid token or valid prefix subject once constructed. Happy to add test for that if we feel we should.

Reply:
I would add the test just so we can fail fast (on startup).

Reply:
but can be separate pr

Reply:
I can add it in..

Reply:
I feel quite strongly we should not allow `.` in a domain name, it significantly complicates writing ACL lists when domains can have dots, if a domain is a single token its really easy to be specific and restrctive, if its multiple tokens how would I write an access list allowing access to just one domain?

It also removes cognitive overhead since there's a single answer to what are the rules for jetstream object names - no dots anywhere. Adding a "but" to that is not good.

Reply:
Done in https://github.com/nats-io/nats-server/pull/2186